internal/provider/alert_resource.go

```diff
@@ -44,112 +44,87 @@ func (r *alertResource) ValidateConfig(ctx context.Context, req resource.Validat
 		return
 	}
 
-	if len(data.Conditions) > 5 {
-		resp.Diagnostics.AddAttributeError(
-			path.Root("conditions"),
-			"More than five conditions.",
-			"Cannot support more than five conditions at this time.",
-		)
-		return
-	} else if len(data.Conditions) < 1 {
-		resp.Diagnostics.AddAttributeError(
-			path.Root("conditions"),
-			"No conditions.",
-			"One or more conditions are required to trigger the alert.",
-		)
-		return
+	resp.Diagnostics.Append(data.validateConditions()...)
+}
+
+func (model *alertResourceModel) validateConditions() []diag.Diagnostic {
```

Comment:
```suggestion
func (model *alertResourceModel) validateConditions() diag.Diagnostics {
```

You can leave the `return` statements below as they are, or change them to:

```go
return diag.Diagnostics{
	diag.NewAttributeErrorDiagnostic(...)
}
```

But this response type needs to be `diag.Diagnostics` if you want to take advantage of the methods defined for this type.